reload/fs.go

```diff
@@ -0,0 +1,60 @@
+package reload
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+)
+
+type FS struct {
+	embed fs.FS
+	dir   fs.FS
+}
+
+func NewFS(embed fs.ReadDirFS, dir string) FS {
+	return FS{
+		embed: embed,
+		dir:   os.DirFS(dir),
+	}
+}
+
+func (f FS) Open(name string) (fs.File, error) {
+	if name == "embed.go" {
+		return nil, fs.ErrNotExist
+	}
+	file, err := f.getFile(name)
+	if name != "." {
+		return file, err
+	}
+	return rootFile{file}, err
+}
+
+func (f FS) getFile(name string) (fs.File, error) {
+	cfgFile := "./.buffalo.dev.yml"
+	if _, err := os.Stat(cfgFile); err != nil {
+		return f.embed.Open(name)
+	}
+	return f.dir.Open(name)
+}
```

Comment:
This is a design decision. How about allowing live access even in production mode outside the application tree? E.g. just try `f.dir.Open` first and do `f.embed.Open` when it failed? What do you think @paganotoni ?

When I developed a small in-house application with buffalo about 4 years ago, it was very useful when I need to update some HTML files, updated images, and auto-response templates without code change.

Reply:
I now changed the implementation to try `f.dir.Open` and only when it fails fall back to `f.embed.Open`. This would then also require the built binary to be run from the same folder as `main.go` for this hot swapping in a deployed environment to work.

I personally hav only ever used a buffalo application within a docker image. So when we update assets we would need to build a new image anyway and have it deployed.

Reply:
Yeah, I agree that is a modern and standard way to deploy services nowadays.